railties/lib/rails/generators/app_base.rb

```diff
@@ -327,59 +324,26 @@ def convert_database_option_for_jruby
       def assets_gemfile_entry
         return [] if options[:skip_sprockets]
 
-        gems = []
-        gems << GemfileEntry.version("sass-rails", "~> 5.0",
-                                     "Use SCSS for stylesheets")
-
-        if !options[:skip_javascript]
-          gems << GemfileEntry.version("uglifier",
-                                     ">= 1.3.0",
-                                     "Use Uglifier as compressor for JavaScript assets")
-        end
-
-        gems
+        GemfileEntry.version("sass-rails", "~> 5.0", "Use SCSS for stylesheets")
       end
 
       def webpacker_gemfile_entry
-        return [] unless options[:webpack]
+        return [] if options[:skip_javascript]
 
-        comment = "Transpile app-like JavaScript. Read more: https://github.com/rails/webpacker"
-        GemfileEntry.new "webpacker", nil, comment
+        GemfileEntry.new "webpacker", nil, "Transpile app-like JavaScript. Read more: https://github.com/rails/webpacker"
       end
 
       def jbuilder_gemfile_entry
         comment = "Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder"
         GemfileEntry.new "jbuilder", "~> 2.5", comment, {}, options[:api]
       end
 
-      def coffee_gemfile_entry
-        GemfileEntry.version "coffee-rails", "~> 4.2", "Use CoffeeScript for .coffee assets and views"
-      end
-
       def javascript_gemfile_entry
-        if options[:skip_javascript] || options[:skip_sprockets]
+        if options[:skip_javascript] || options[:skip_turbolinks]
           []
         else
-          gems = [javascript_runtime_gemfile_entry]
-          gems << coffee_gemfile_entry unless options[:skip_coffee]
-
-          unless options[:skip_turbolinks]
-            gems << GemfileEntry.version("turbolinks", "~> 5",
-             "Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks")
-          end
-
-          gems
-        end
-      end
-
-      def javascript_runtime_gemfile_entry
-        comment = "See https://github.com/rails/execjs#readme for more supported runtimes"
-        if defined?(JRUBY_VERSION)
-          GemfileEntry.version "therubyrhino", nil, comment
-        elsif RUBY_PLATFORM =~ /mingw|mswin/
-          GemfileEntry.version "duktape", nil, comment
-        else
-          GemfileEntry.new "mini_racer", nil, comment, { platforms: :ruby }, true
+          [ GemfileEntry.version("turbolinks", "~> 5",
+             "Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks") ]
```

Comment:
Turbolinks is added on package.json, so it looks unnecessary to add turbolinks to Gemfile.

Reply:
I see. Thanks. https://github.com/turbolinks/turbolinks-rails